server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTarget.java

```diff
@@ -148,8 +147,9 @@ public void startReplication(ActionListener<Void> listener) {
         cancellableThreads.setOnCancel((reason, beforeCancelEx) -> {
             // This method only executes when cancellation is triggered by this node and caught by a call to checkForCancel,
             // SegmentReplicationSource does not share CancellableThreads.
-            final CancellableThreads.ExecutionCancelledException executionCancelledException =
-                new CancellableThreads.ExecutionCancelledException("replication was canceled reason [" + reason + "]");
+            final ReplicationFailedException executionCancelledException = new ReplicationFailedException(
+                "replication was canceled reason [" + reason + "]"
+            );
             notifyListener(executionCancelledException, false);
```

Comment:
With notifyListener now accepting a ReplicationFailedException, we should wrap the cancellationException in ReplicationFailedException. 

```java
        cancellableThreads.setOnCancel((reason, beforeCancelEx) -> {
            // This method only executes when cancellation is triggered by this node and caught by a call to checkForCancel,
            // SegmentReplicationSource does not share CancellableThreads.
            final CancellableThreads.ExecutionCancelledException executionCancelledException =
                new CancellableThreads.ExecutionCancelledException("replication was canceled reason [" + reason + "]");
            notifyListener(new ReplicationFailedException("Segment replication failed to complete", executionCancelledException), false);
            throw executionCancelledException;
        });
```

There is some special logic above in SegmentReplicationTarget that overrides notifyListener to unwrap the failure and mark it as cancelled.  This block will never trigger if the cancellation initiates from the primary however.  So thats why the special logic is not in this setOnCancel block.

Reply:
ahh, thanks for the help. For some reason I was not able to figure the trigger out exactly